packages/react-data-grid/src/Canvas.tsx

```diff
@@ -260,49 +267,70 @@ export default class Canvas<R> extends React.PureComponent<CanvasProps<R>> {
     return <Row<R> {...props} />;
   }
 
-  renderPlaceholder(key: string, height: number) {
-    return (
-      <div key={key} style={{ height }} />
-    );
+  mapToRowElement = ({ row, subRowDetails }: RowRendererData<R>, idx: number): JSX.Element => {
+    const {
+      rowOverscanStartIdx,
+      cellMetaData,
+      columns,
+      colOverscanStartIdx,
+      colOverscanEndIdx,
+      lastFrozenColumnIndex,
+      rowHeight,
+      pinnedRows,
+      rowsCount
+    } = this.props;
+
+    const isBottomPinned = pinnedRows && pinnedRows.includes(row);
+
+    const rowIdx = isBottomPinned ? rowsCount + 1 + idx : rowOverscanStartIdx + idx;
+
+    return row && this.renderRow({
+      key: rowIdx,
+      ref: (row: (RowRenderer<R> & React.Component<RowRendererProps<R>>) | null) => {
+        if (row) {
+          this.rows.set(rowIdx, row);
+        } else {
+          this.rows.delete(rowIdx);
+        }
+      },
+      idx: rowIdx,
+      row,
+      height: rowHeight,
+      columns,
+      isSelected: this.isRowSelected(rowIdx, row),
+      cellMetaData,
+      subRowDetails,
+      colOverscanStartIdx,
+      colOverscanEndIdx,
+      lastFrozenColumnIndex,
+      isScrolling: this.props.isScrolling,
+      scrollLeft: this.props.scrollLeft,
+      isBottomPinned
+    });
   }
 
   render() {
-    const { rowOverscanStartIdx, rowOverscanEndIdx, cellMetaData, columns, colOverscanStartIdx, colOverscanEndIdx, colVisibleStartIdx, colVisibleEndIdx, lastFrozenColumnIndex, rowHeight, rowsCount, width, height, rowGetter, contextMenu } = this.props;
+    const { scrollTop, rowOverscanStartIdx, rowOverscanEndIdx, columns, colVisibleStartIdx, colVisibleEndIdx, rowHeight, rowsCount, width, height, rowGetter, contextMenu, pinnedRows } = this.props;
     const RowsContainer = this.props.RowsContainer || RowsContainerDefault;
 
-    const rows = this.getRows(rowOverscanStartIdx, rowOverscanEndIdx)
-      .map(({ row, subRowDetails }, idx) => {
-        const rowIdx = rowOverscanStartIdx + idx;
-        return row && this.renderRow({
-          key: rowIdx,
-          ref: (row: (RowRenderer<R> & React.Component<RowRendererProps<R>>) | null) => {
-            if (row) {
-              this.rows.set(rowIdx, row);
-            } else {
-              this.rows.delete(rowIdx);
-            }
-          },
-          idx: rowIdx,
-          row,
-          height: rowHeight,
-          columns,
-          isSelected: this.isRowSelected(rowIdx, row),
-          cellMetaData,
-          subRowDetails,
-          colOverscanStartIdx,
-          colOverscanEndIdx,
-          lastFrozenColumnIndex,
-          isScrolling: this.props.isScrolling,
-          scrollLeft: this.props.scrollLeft
-        });
-      });
+    const pinnedRowsHeight = isIEOrEdge() && pinnedRows ? pinnedRows.length * rowHeight : 0;
```

Comment:
IE Edge horizontal sticky is not working during horizontal scroll if the outside container is sticky as well.

![IE edge sticky failure for pinned rows](https://user-images.githubusercontent.com/25043398/66163904-054b1400-e5f7-11e9-88ba-f238944b0259.gif)